sadu-updater/src/main/java/de/chojo/sadu/updater/SqlUpdater.java

```diff
@@ -127,30 +125,33 @@ private SqlUpdater(DataSource source, QueryBuilderConfig config, String versionT
      * @throws IOException if the version file does not exist.
      */
     @CheckReturnValue
-    public static <T extends JdbcConfig<?>> SqlUpdaterBuilder<T> builder(DataSource dataSource, Database<T> type) throws IOException {
+    public static <T extends JdbcConfig<?>, U extends UpdaterBuilder<T, ?>> U builder(DataSource dataSource, Database<T, U> type) throws IOException {
         var version = "";
         try (var versionFile = SqlUpdater.class.getClassLoader().getResourceAsStream("database/version")) {
             version = new String(versionFile.readAllBytes(), StandardCharsets.UTF_8).trim();
         }
 
         var ver = version.split("\\.");
-        return new SqlUpdaterBuilder<>(dataSource, new SqlVersion(Integer.parseInt(ver[0]), Integer.parseInt(ver[1])), type);
+        return builder(dataSource, new SqlVersion(Integer.parseInt(ver[0]), Integer.parseInt(ver[1])), type);
     }
 
     /**
-     * Creates a new {@link SqlUpdaterBuilder} with a version set to a string located in {@code resources/database/version}.
+     * Creates a new {@link BaseSqlUpdaterBuilder} with a version set to a string located in {@code resources/database/version}.
      *
      * @param dataSource the data source to connect to the database
      * @param version    the version with {@code Major.Patch}
      * @param type       the sql type of the database
      * @param <T>        type of the database defined by the {@link Database}
      * @return builder instance
      */
-    public static <T extends JdbcConfig<?>> SqlUpdaterBuilder<T> builder(DataSource dataSource, SqlVersion version, Database<T> type) {
-        return new SqlUpdaterBuilder<>(dataSource, version, type);
+    public static <T extends JdbcConfig<?>, U extends UpdaterBuilder<T, ?>> U builder(DataSource dataSource, SqlVersion version, Database<T, U> type) {
+        var builder = type.newSqlUpdaterBuilder();
+        builder.setSource(dataSource);
+        builder.setVersion(version);
+        return (U) builder;
     }
 
-    private void init() throws IOException, SQLException {
+    public void init() throws IOException, SQLException {
```

Comment:
Should be annotated with internal, since calling this might have unexpected behaviour.